test_rclcpp/CMakeLists.txt

```diff
@@ -198,6 +204,44 @@ if(AMENT_ENABLE_TESTING)
         set_tests_properties(parameter_server_cpp__${middleware_impl} PROPERTIES DEPENDS "test_parameters_server_cpp__${middleware_impl} test_remote_parameters_cpp__${middleware_impl}")
     endif()
 
+    # Service tests single implementation
+    add_executable(test_services_client_cpp__${middleware_impl} "test/test_services_client.cpp")
+    target_link_libraries(test_services_client_cpp__${middleware_impl} ${GTEST_LIBRARIES})
+    # the above could be replaced by a helper which won't register the gtest for running
+    # ament_add_gtest_subtest(test_services_client_cpp__${middleware_impl} "test/test_services_client.cpp")
+    target_include_directories(test_services_client_cpp__${middleware_impl} PUBLIC
+      "${GTEST_INCLUDE_DIRS}")
+    add_dependencies(test_services_client_cpp__${middleware_impl} ${PROJECT_NAME})
+    rosidl_target_interfaces(test_services_client_cpp__${middleware_impl}
+      ${PROJECT_NAME} ${typesupport_impl})
+    ament_target_dependencies(test_services_client_cpp__${middleware_impl}
+      "${middleware_impl}"
+      "rclcpp")
+    add_executable(test_services_server_cpp__${middleware_impl} "test/test_services_server.cpp")
+    add_dependencies(test_services_server_cpp__${middleware_impl} ${PROJECT_NAME})
+    rosidl_target_interfaces(test_services_server_cpp__${middleware_impl}
+      ${PROJECT_NAME} ${typesupport_impl})
+    ament_target_dependencies(test_services_server_cpp__${middleware_impl}
+      "${middleware_impl}"
+      "rclcpp")
+    # TODO consider using a generator expression instead
+    # $<TARGET_FILE:target>
+    # until then set the policy explicitly in order to avoid warning with newer CMake versions.
+    if(POLICY CMP0026)
```

Comment:
I'm going to run this on Windows and check if this code works, otherwise I'll set the paths to the executables manually.